InfiniteRecharge/src/main/java/frc/robot/subsystem/Subsystems.java

```diff
@@ -0,0 +1,14 @@
+package frc.robot.subsystem;
+
+
+
+public enum Subsystems {
+    CLIMB,
```

Comment:
the idea behind it was that we could just call one method per subsystem (addSubsystem) in a very general way to reduce the work we do when adding dependencies and subsystems. Instead of saying, for example, driveSubsystem = new DriveSubsystem(climbSubsystem, navigationSusbsytem, ...) we could just say addSubsystem(new DriveSubsystem()) and all dependencies automatically without us having to specify which ones ourselves. But by removing us using variables and therefore identifying subsystems by a name, it would require some other way to identify which object is which subsystem. That's what this class is for - every subsystem declares which one it is, and the rest can just find the DRIVE subsystem and then find the corresponding object. Am I making any sense?

Reply:
Yeah, I get the idea. I still think we should explicitly create each Subsystem in the robot class, rather than trying to write our own dependency injection system. If we want to use DI, we should just add a dependency to google guice and use it.

If you really want to keep a map of classes by type, you can use the Class object as the type.

```java
        Map<Class<? extends Subsystem>, Subsystem> subsystemsByType = new HashMap<>();

        subsystemsByType.put(TalonDriveSubsystem.class, new TalonDriveSubsystem(config));
        subsystemsByType.put(SparkDriveSubsystem.class, new SparkDriveSubsystem(config));

        SparkDriveSubsystem ss = (SparkDriveSubsystem) subsystemsByType.get(SparkDriveSubsystem.class);

```

Reply:
yeah that would also work. Now that I think about it more we could also just have an array[list] and use the names of each subsystem (the ones in setName()) to ID them